lib/galaxy/authnz/psa_authnz.py

```diff
@@ -179,7 +179,11 @@ def refresh(self, trans, user_authnz_token):
         else:
             log.debug("No `expires` or `expires_in` key found in token extra data, cannot refresh")
             return False
-        if int(user_authnz_token.extra_data["auth_time"]) + int(expires) / 2 <= int(time.time()):
+        if (
+            int(user_authnz_token.extra_data["auth_time"]) + int(expires) / 2
```

Comment:
if the `expires` is "seconds to expire from issue time" then this comparison is broken I think

Reply:
This is how I understand it: if A (auth time + half lifetime), B (now) and C (expiration time) are specific instants of time, that's just "renew when `A ≤ B < C`", in other words, "renew when now is later than auth time + half lifetime and less than expiration time.

Did I miss something there? Everything else is also in seconds right?

Reply:
I think you're correct.